jgiven-gradle-plugin/src/test/java/com/tngtech/jgiven/gradle/JGivenPluginTest.java

```diff
@@ -62,111 +64,112 @@ public void plugin_can_build_with_an_empty_project() throws Exception {
-                + "  }\n"
-                + "}\n")
+                .the_plugin_is_applied()
+                .and().there_are_JGiven_tests()
+                .and().the_jgiven_report_is_configured_by("reports {\n"
+                        + "  html {\n"
+                        + "    required = false\n"
+                        + "  }\n"
+                        + "  text {\n"
+                        + "    required = true\n"
+                        + "  }\n"
+                        + "}\n")
         ;
 
         when()
-            .a_build().with()
-            .the_task("test").and()
-            .the_task("jgivenTestReport").is_successful();
+                .a_build().with()
+                .the_task("test").and()
+                .the_task("jgivenTestReport").is_successful();
 
         then()
-            .the_JGiven_reports_are_not_written_to("build/reports/jgiven/test/html")
-            .and().the_JGiven_text_reports_are_written_to("build/reports/jgiven/test/text");
+                .the_JGiven_reports_are_not_written_to("build/reports/jgiven/test/html")
+                .and().the_JGiven_text_reports_are_written_to("build/reports/jgiven/test/text");
 
     }
 
     @Test
     public void configure_html_report() throws IOException {
         given()
-            .the_plugin_is_applied()
-            .and().there_are_JGiven_tests()
-            .and().the_jgiven_report_is_configured_by("reports {\n"
-                + "  html {\n"
-                + "    required = true\n"
-                + "    title = 'JGiven Gradle Plugin'\n"
-                + "  }\n"
-                + "}\n")
+                .the_plugin_is_applied()
+                .and().there_are_JGiven_tests()
+                .and().the_jgiven_report_is_configured_by("reports {\n"
+                        + "  html {\n"
+                        + "    required = true\n"
+                        + "    title = 'JGiven Gradle Plugin'\n"
+                        + "  }\n"
+                        + "}\n")
         ;
 
         when()
-            .a_build().with()
-            .the_task("test").and()
-            .the_task("jgivenTestReport").is_successful();
+                .a_build().with()
+                .the_task("test").and()
+                .the_task("jgivenTestReport").is_successful();
 
         then()
-            .the_JGiven_html_reports_are_written_to("build/reports/jgiven/test/html");
+                .the_JGiven_html_reports_are_written_to("build/reports/jgiven/test/html");
 
     }
 
     @Test
     void no_jgiven_results_no_report() throws IOException {
 
         given()
-            .the_plugin_is_applied()
-            .and().there_are_JGiven_tests();
+                .the_plugin_is_applied()
+                .and().there_are_JGiven_tests();
 
         when()
-            .a_build().with()
-            .the_task("jgivenTestReport").is_successful();
+                .a_build().with()
+                .the_task("-x").the_task("test").and()
```

Comment:
@jjohannes I'd like to, for one final time, have your opinon on this one. 
To me it seems, now that we properly use Providers, Gradle is able to track the  necessary inputs of the JGiven task. 
Therefore, previously a JGiven report task would not run, if a test task was not explicitely requested. Now, I assume that because we explicitely request the report task to run, Gradle knows that it needs to run  the test task and does so accordingly.

While this is changed behavior, I am of the opinon, that we are now more in line with how Gradle is expected to behave and therefore this would generally be an appreciated change.

How do you see this?

Reply:
Yes, your observation is correct. The providers that point at task outputs carry the task dependency information and that is why the `test` task is automatically executed before.

If you can accept the breaking change, I would agree that this is the better behavior. It follows Gradle best practices. (If not, there would be ways to "trick" the setup into loosing the dependency but still using providers in general.)

Note: Traditionally the `jacoco` report task built into Gradle core also does not follow this best practice. However I believe this is due to not wanting to break the behavior that was introduced early (pre Gradle 1.0). If the JaCoCo tasks/plugin would be created today, I think it would also behave different.